libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/Tracker.java

```diff
@@ -0,0 +1,83 @@
+package org.wordpress.android.analytics;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+
+import java.util.Map;
+import java.util.UUID;
+
+import org.wordpress.android.analytics.AnalyticsTracker.Stat;
+import org.wordpress.android.util.AppLog;
+
+public abstract class Tracker {
+    abstract void track(Stat stat);
+    abstract void track(Stat stat, Map<String, ?> properties);
+    abstract void endSession();
+    abstract void refreshMetadata(boolean isUserConnected,boolean isWordPressComUser, boolean isJetpackUser,
+                         int sessionCount, int numBlogs, int versionCode, String username, String email);
+    abstract void registerPushNotificationToken(String regId);
+    abstract String getAnonIdPrefKey();
+
+    private String mAnonID = null; // do not access this variable directly. Use methods.
+    private String mWpcomUserName = null;
+    Context mContext;
+
+    public Tracker(Context context) {
+        if (null == context) {
```

Comment:
Since `mContext` is required by so much of the code, it would make sense to throw an `IllegalArgumentException` exception if a null context is passed.